src/BenchmarkDb/Program.cs

```diff
@@ -0,0 +1,262 @@
+                        var now = DateTime.UtcNow;
+                        var tps = (int) (_counter / (now - lastDisplay).TotalSeconds);
+                        var remaining = (int) (time - (now - startTime).TotalSeconds);
+                        listOfResults.Add(tps);
+
+                        Console.Write($"{tps} tps, {remaining}s                     ");
+                        Console.SetCursorPosition(0, Console.CursorTop);
+                        lastDisplay = now;
+                        totalTransactions += Interlocked.Exchange(ref _counter, 0);
+                    }
+                })
+            );
+
+            // Signals the beginning and the end of the test
+            tasks.Add(
+                Task.Run(async () =>
+                {
+                    await Task.Delay(TimeSpan.FromSeconds(time + Warmup));
+
+                    Interlocked.Exchange(ref _stopping, 1);
+                    stopTime = DateTime.UtcNow;
+                })
+            );
+
+            Task.WhenAll(tasks).GetAwaiter().GetResult();
+
+            var totalTps = (int) (totalTransactions / (stopTime - startTime).TotalSeconds);
+
+
+            listOfResults.Remove(listOfResults.Max());
+            listOfResults.Remove(listOfResults.Min());
+            var stddev = CalculateStdDev(listOfResults, listOfResults.Count);
+
+            Console.SetCursorPosition(0, Console.CursorTop);
+            Console.WriteLine($"{numThreads:D2} Threads, tps: {totalTps:F2}, stddev(w/o best+worst): {stddev:F2}");
+
+            var desc = $"{database}+{mode}+{numThreads}";
+
+            using (var sw = File.AppendText("results.md"))
+            {
+                sw.WriteLine($"|{desc}|{mode}|{numThreads:D2}|{totalTps:F0}|{stddev:F0}|");
+            }
+
+            using (var sw = File.AppendText("results.csv"))
+            {
+                sw.WriteLine($"{desc},{mode},{numThreads:D2},{totalTps:F0},{stddev:F0}");
+            }
+
+            double CalculateStdDev(IEnumerable<double> values, int count)
+            {
+                var avg = values.Sum() / count;
+                double sum = values.Sum(d => Math.Pow(d - avg, 2));
+                return Math.Sqrt((sum) / count);
+            }
+
+            async Task DoWorkAsync()
+            {
+                while (_stopping != 1)
+                {
+                    Interlocked.Increment(ref _counter);
+
+                    try
+                    {
+                        var results = new List<Fortune>();
+                        using (var connection = factory.CreateConnection())
+                        {
+                            connection.ConnectionString = connectionString;
+                            await connection.OpenAsync();
+
+                            using (var command = connection.CreateCommand())
+                            {
+                                command.CommandText = SqlQuery;
+                                command.Prepare();
```

Comment:
Can we move connection and command initialization outside the loop? I.e. Cache connection and command per thread.

Reply:
Is that exactly what @damageboy's KeepConn/KeepCmd variations do?

Reply:
Not sure, but this is what my variations do :smile:

Reply:
See https://github.com/damageboy/benchmarkdb/blob/master/BenchmarkDb/Program.cs#L215 for example.

Reply:
I let it explicitly in the loop to match what we do in TE Fortunes, per request. Otherwise as @divega mentioned it would be the conn+cmd variation from @damageboy

Reply:
Question then is do we want a mode that reuses the command and the connection? Do we care about it? I assume we could, as it would give more information about the cost of these.

Reply:
Agree we want this